docs/frontend.asciidoc

```diff
@@ -0,0 +1,108 @@
+[[frontend]]
+== Frontend support
+
+NOTE: Frontend support is an experimental feature not intended for production usage at the moment. 
+
+This section describes features specifically designed to support frontend application monitoring
+and a guide on how to enable experimental frontend support. 
+
+[[frontend-enable]]
+[float]
+=== Enable Frontend Support
+To try out experimental frontend support, set the `apm-server.frontend.enabled` to `true`.
+See https://github.com/elastic/apm-server/blob/{doc-branch}/apm-server.reference.yml[`apm-server.reference.yml`] for more configuration options.
+
+[[sourcemap]]
+=== Source Maps
+It is common practice to compress and minify client side javascript code for several reasons, e.g. performance gain. 
+This can make debugging very difficult, as it is hard to read the minified files.
+Source mapping can help for debugging minfied javascript files, 
+by mapping code from the minified files to the original source code. 
+
+APM Server frontend support has a source map <<sourcemap-endpoint,upload endpoint>> 
+form uploading source maps complying to the 
```

Comment:
`APM Server has a source map [upload endpoint] which accepts source maps complying to the [source map revision 3 proposal]